mmv1/templates/terraform/custom_flatten/image_kms_key_name.go.erb

```diff
@@ -0,0 +1,21 @@
+<%# The license inside this block applies to this file.
+	# Copyright 2022 Google Inc.
+	# Licensed under the Apache License, Version 2.0 (the "License");
+	# you may not use this file except in compliance with the License.
+	# You may obtain a copy of the License at
+	#
+	#     http://www.apache.org/licenses/LICENSE-2.0
+	#
+	# Unless required by applicable law or agreed to in writing, software
+	# distributed under the License is distributed on an "AS IS" BASIS,
+	# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	# See the License for the specific language governing permissions and
+	# limitations under the License.
+-%>
+func flatten<%= prefix -%><%= titlelize_property(property) -%>(v interface{}, d *schema.ResourceData, config *Config) interface{} {
+	if v == nil {
+		return v
+	}
+	vStr := v.(string)
+	return strings.Split(vStr, "/cryptoKeyVersions/")[0]
```

Comment:
If I'm understanding this right, the API's returning a KMS key version when we're providing a key? Can you file a bug against the API to figure out why?

Their documentation states:

> The name of the encryption key that is stored in Google Cloud KMS. For example: "kmsKeyName": "projects/ kms_project_id/locations/ region/keyRings/key_region/cryptoKeys/key

Reply:
Done

Reply:
Bug is b/261642503, for posterity. No response yet.

We should be safe against changes to the format, since we're checking `[0]` after the split. Even if the API changes to the documented format we'd receive the same result. This is non-blocking, due to that.